l10n_es_ticketbai_api_batuz/models/lroe_operation.py

```diff
@@ -0,0 +1,507 @@
+    subchapter_pf_140_7_3_full = "7.3 - Transmisiones de inmuebles sujetas a IVA"
+    subchapter_pf_140_7_4 = "7.4"
+    subchapter_pf_140_7_4_full = (
+        "7.4 - Importes superiores a 6.000 euros percibidos en metálico"
+    )
+    chapter_pf_140_8 = "8"
+    chapter_pf_140_8_full = "8 - Criterio de caja / Criterio de cobros y pagos"
+    subchapter_pf_140_8_1 = "8.1"
+    subchapter_pf_140_8_1_full = "8.1 - Alta de agrupaciones de bienes"
+    subchapter_pf_140_8_2 = "8.2"
+    subchapter_pf_140_8_2_full = "8.2 - Baja de agrupaciones de bienes"
+
+
+class LROEOperationEnum(Enum):
+    create = "A00"
+    cancel = "AN0"
+    update = "M00"
+    query = "C00"
+
+
+class LROEOperationStateEnum(Enum):
+    DRAFT = "draft"
+    ERROR = "error"
+    CANCEL = "cancel"
+    RECORDED_WARNING = "recorded_warning"
+    RECORDED = "recorded"
+
+
+class LROEOperationVersion(Enum):
+    v1 = "1.0"
+
+
+class LROEChapter(models.Model):
+    _name = "lroe.chapter"
+    _description = "LROE Chapter"
+    _order = "model, code, id"
+
+    name = fields.Char(required=True)
+    code = fields.Char(required=True)
+    model = fields.Selection(
+        [
+            (LROEModelEnum.model_pj_240.value, "LROE PJ 240"),
+            (LROEModelEnum.model_pf_140.value, "LROE PF 140"),
+        ],
+        string="LROE Model",
+        required=True,
+        default=LROEModelEnum.model_pj_240.value,
+    )
+    lroe_chapter_id = fields.Many2one(
+        comodel_name="lroe.chapter", string="Parent chapter"
+    )
+
+
+class LROEOperation(models.Model):
+    _name = "lroe.operation"
+    _description = "LROE Operation"
+    _order = "id desc"
+
+    name = fields.Char(compute="_compute_lroe_operation_name", store=True)
+    company_id = fields.Many2one("res.company", required=True)
+    type = fields.Selection(
+        [
+            (LROEOperationEnum.create.value, "Create"),
+            (LROEOperationEnum.cancel.value, "Cancel"),
+            (LROEOperationEnum.update.value, "Update"),
+            (LROEOperationEnum.query.value, "Query"),
+        ],
+        required=True,
+        default=LROEOperationEnum.create.value,
+    )
+    model = fields.Selection(
+        "Model", related="company_id.lroe_model", readonly=True, required=True
+    )
+    state = fields.Selection(
+        selection=[
+            (LROEOperationStateEnum.DRAFT.value, "Draft"),
+            (LROEOperationStateEnum.ERROR.value, "Error"),
+            (LROEOperationStateEnum.CANCEL.value, "Cancel"),
+            (LROEOperationStateEnum.RECORDED_WARNING.value, "Recorded Warning"),
+            (LROEOperationStateEnum.RECORDED.value, "Recorded"),
+        ],
+        default=LROEOperationStateEnum.DRAFT.value,
+    )
+    tbai_invoice_ids = fields.One2many(
+        comodel_name="tbai.invoice",
+        inverse_name="lroe_operation_id",
+        string="TBai Customer Invoices",
+    )
+    xml_datas = fields.Binary()
+    xml_datas_fname = fields.Char("XML File Name")
```

Comment:
xml_datas_fname y trx_gzip_fname tiene el mismo nombre XML File Name, tiene sentido, esto arroja un warnin en odoo.

2023-05-30 20:32:16,178 255 WARNING odoo odoo.addons.base.models.ir_model: Two fields (trx_gzip_fname, xml_datas_fname) of lroe.operation() have the same label: XML File Name. [Modules: l10n_es_ticketbai_api_batuz and l10n_es_ticketbai_api_batuz]